scripts/download_osm_data.py

```diff
@@ -425,338 +69,48 @@ def convert_iso_to_geofk(iso_code, iso_coding=True, convert_dict=iso_to_geofk_di
-
-        # extract countries in continent
-        elif value1 in world_iso.keys():
-            codes_list = list(world_iso[value1])
-
-        # extract countries in regions
-        elif value1 in continent_regions.keys():
-            codes_list = continent_regions[value1]
-
-        # extract countries
-        else:
-            codes_list.extend([value1])
-
-        # create a list with all countries
-        full_codes_list.extend(codes_list)
-
-    # Removing duplicates and filter outputs by coding
-    full_codes_list = filter_codes(list(set(full_codes_list)), iso_coding=iso_coding)
-
-    return full_codes_list
-
-
-def country_list_to_geofk(country_list):
-    """
-    Convert the requested country list into geofk norm
-
-    Parameters
-    ----------
-    input : str
-        Any two-letter country name or aggregation of countries given in config_osm_data.py
-        Country name duplications won't distort the result.
-        Examples are:
-        ["NG","ZA"], downloading osm data for Nigeria and South Africa
-        ["SNGM"], downloading data for Senegal&Gambia shape
-        ["NG","ZA","NG"], won't distort result.
-
-    Returns
-    -------
-    full_codes_list : list
-        Example ["NG","ZA"]
-    """
-
-    full_codes_list = [convert_iso_to_geofk(c_code) for c_code in set(country_list)]
-
-    return full_codes_list
-
-
 if __name__ == "__main__":
     if "snakemake" not in globals():
-        from _helpers import mock_snakemake
+        from _helpers import mock_snakemake, sets_path_to_root
 
         os.chdir(os.path.dirname(os.path.abspath(__file__)))
         snakemake = mock_snakemake("download_osm_data")
+        sets_path_to_root("pypsa-earth")
     configure_logging(snakemake)
 
-    # Required to set path to pypsa-earth
-    sets_path_to_root("pypsa-earth")
-
-    # ["substation", "generator", "line", "cable", "tower"]
-    feature_list = ["substation", "generator", "line", "cable"]
-
-    # get list of countries into geofabrik convention; expected iso norm in input
+    store_path_data = Path.joinpath(Path().cwd(), "data", "osm")
+    store_path_resources = Path.joinpath(Path().cwd(), "resources", "osm", "raw")
     country_list = country_list_to_geofk(snakemake.config["countries"])
-    output_files = snakemake.output  # output snakemake
-    nprocesses = snakemake.config.get(
-        "download_osm_data_nprocesses", 1
-    )  # number of threads
-
-    # get the default and metric crs of the workflow
-    geo_crs = snakemake.config["crs"]["geo_crs"]
-    distance_crs = snakemake.config["crs"]["distance_crs"]
 
-    # Set update # Verify = True checks local md5s and pre-filters data again
-    process_data(
-        feature_list,
-        country_list,
-        output_files,
-        geo_crs,
-        distance_crs,
-        iso_coding=True,
+    # Python interface to download OpenStreetMap data
+    # Documented at https://github.com/pypsa-meets-earth/earth-osm
+    eo.get_osm_data(
```

Comment:
may it be worthy to encapsulate this function in an additional function that does the filtering by the iso_to_geofk function?
This may be handy for other rules e.g. -sec or -distribution

Reply:
Great idea. Let's do that when needed? (probably takes a while until it is necessary)